activerecord/test/cases/scoping/default_scoping_test.rb

```diff
@@ -179,6 +179,23 @@ def test_default_scope_with_all_queries_runs_on_update_columns
     assert_match(/mentor_id/, update_sql)
   end
 
+  def test_default_scope_with_all_queries_is_applied_once_on_update_columns
+    Mentor.create!
+    dev = DeveloperWithDefaultMentorScopeAllQueries.create!(name: "Eileen")
+    relation = DeveloperWithDefaultMentorScopeAllQueries.where("1=1")
+    update_sql = capture_sql { relation.scoping(all_queries: true) { dev.update_columns(name: "Not Eileen") } }.first
+
+    assert_match(/WHERE .\w+.\..id. = \S+ AND .\w+.\..mentor_id. = \S+ AND \(1=1\)\Z/, update_sql)
```

Comment:
I couldn't find a better way to test that `mentor_id` only appeared once. I tried this with sqlite, mysql and postgresql, since they all differ slightly in how they quote table and column names as well as what they pass on the right hand side, `?`, the actual value, `$1`, respectively. Happy to hear suggestions for a less obscure version.

Reply:
>since they all differ slightly in how they quote table 

This can often be solved by building a dynamic expectation, for example
https://github.com/rails/rails/blob/b9a5ba46dd162fde12e638ea6f34cfb29aa76b3e/activerecord/test/cases/associations/eager_test.rb#L510-L515

Maybe we can be as straightforward and just count `mentor_id` appearances in the sql. We know that `mentor_id` can only be used here as a column name, it won't be the column we are updating nor the new value so perhaps the assertion can be `assert_equal(1, sql.scan("mentor_id").size)`